src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -91,6 +92,7 @@ def __init__(self, mgr: "CephadmOrchestrator"):
             self.upgrade_state: Optional[UpgradeState] = UpgradeState.from_json(json.loads(t))
         else:
             self.upgrade_state = None
+        self.upgrade_completion_message = ''  # for mypy
```

Comment:
not just for mypy. it is good practice to declare all members in `__init__`

```suggestion
        self.upgrade_completion_message = ''
```